src/secret-storage.ts

```diff
@@ -146,3 +150,340 @@ export interface SecretStorageCallbacks {
         name: string,
     ) => Promise<[string, Uint8Array] | null>;
 }
+
+interface SecretInfo {
+    encrypted: {
+        [keyId: string]: IEncryptedPayload;
+    };
+}
+
+interface Decryptors {
+    encrypt: (plaintext: string) => Promise<IEncryptedPayload>;
+    decrypt: (ciphertext: IEncryptedPayload) => Promise<string>;
+}
+
+/**
+ * Implementation of Server-side secret storage.
+ *
+ * Secret *sharing* is *not* implemented here: this class is strictly about the storage component of
+ * SSSS.
+ *
+ * @see https://spec.matrix.org/v1.6/client-server-api/#storage
+ */
+export class SecretStorage {
```

Comment:
It's highly regrettable that this gives us two different classes called `SecretStorage`. On the other hand, I can't rename the existing one without breaking backwards compat, and I can't think of a better name for the new one. Suggestions welcome.